pkg/nodeutil/stats_info.go

```diff
@@ -0,0 +1,333 @@
+	currNodeIP, port, err := getIPPort(svr.GetAddr())
+	if err != nil {
+		return nil, err
+	}
+	var si sysinfo.SysInfo
+	// TODO: may be we can impl get cpu, disk, netcard sys info.
+	si.GetSysInfo()
+	cpuInfo := &CPUInfo{
+		CPUVendor:    si.CPU.Vendor,
+		CPUModel:     si.CPU.Model,
+		CPUFrequency: si.CPU.Speed,
+		CPUThread:    si.CPU.Threads,
+	}
+	diskInfos := make([]*DiskInfo, 0, len(si.Storage))
+	for _, v := range si.Storage {
+		diskInfos = append(diskInfos, &DiskInfo{
+			DiskType: v.Driver,
+			DiskSize: v.Size,
+		})
+	}
+	netcardInfos := make([]*NetcardInfo, 0, len(si.Network))
+	for _, v := range si.Network {
+		netcardInfos = append(netcardInfos, &NetcardInfo{
+			Name:       v.Name,
+			Driver:     v.Driver,
+			MACAddress: v.MACAddress,
+			Port:       v.Port,
+			Speed:      v.Speed,
+		})
+	}
+	return &StaticNodeInfo{
+		Name:         svr.Name(),
+		IP:           currNodeIP,
+		Port:         port,
+		CPUInfo:      cpuInfo,
+		MemSize:      si.Memory.Size,
+		DiskInfos:    diskInfos,
+		NetcardInfos: netcardInfos,
+	}, nil
+}
+
+// GetCurrStatsInfo get current stats info
+func GetCurrStatsInfo(svr *server.Server) (interface{}, error) {
+	cpuPercent, err := GetCPUUsage()
+	if err != nil {
+		return nil, err
+	}
+	memPercent, err := GetMemUsage()
+	if err != nil {
+		return nil, err
+	}
+	netInfos, err := GetNetInfo()
+	if err != nil {
+		return nil, err
+	}
+	netUsages := make([]*NetUsage, 0, len(netInfos))
+	for _, v := range netInfos {
+		netUsages = append(netUsages, &NetUsage{
+			Name:      v.Name,
+			BytesSent: v.BytesSent,
+			BytesRecv: v.BytesRecv,
+		})
+	}
+	currNodeIP, _, err := getIPPort(svr.GetAddr())
+	if err != nil {
+		return nil, err
+	}
+	return &StatsInfo{
+		IP:       currNodeIP,
+		Name:     svr.Name(),
+		CPUUsage: cpuPercent,
+		MemUsage: memPercent,
+		NetUsage: netUsages,
+	}, nil
+}
+
+// GetNetworkLatency get network latency
+func GetNetworkLatency(svr *server.Server) (interface{}, error) {
+	currNodeIP, _, err := getIPPort(svr.GetAddr())
+	if err != nil {
+		return "", nil
+	}
+	allNodeIPs := getAllTargetNodeIPs(svr)
+	nls := make([]*NetworkLatency, 0, len(allNodeIPs))
+	for _, ip := range allNodeIPs {
+		if ip == currNodeIP {
+			continue
+		}
+		pingLatency, err := GetNetLatency(ip)
+		if err != nil {
+			log.Println("get target latency failed, error:", err)
+			continue
+		}
+
+		nl := &NetworkLatency{
+			Source:  currNodeIP,
+			Target:  ip,
+			Latency: string(pingLatency),
+			Name:    svr.Name(),
+		}
+		nls = append(nls, nl)
+	}
+	return nls, nil
+}
+
+// getAllTargetNodeIPs get all node ip
+func getAllTargetNodeIPs(svr *server.Server) []string {
+	existNodeHost := map[string]struct{}{}
+	var nodeIPs []string
+	nodes := getPDNode(svr, existNodeHost)
+	if len(nodes) > 0 {
+		nodeIPs = append(nodeIPs, nodes...)
+	}
+	nodes = getTiKVNode(svr, existNodeHost)
+	if len(nodes) > 0 {
+		nodeIPs = append(nodeIPs, nodes...)
+	}
+	nodes = getTiDBNode(svr, existNodeHost)
```

Comment:
I don't like introducing too many TiDB codes into PD. 

What do you think @disksing

Reply:
Agree. Why not send all server IPs from the client?

I think there is some confusion at the abstract level. This PR seems to want to query the status of a single node (including CPU, memory, disk, and network connectivity to other nodes). I assume that all nodes in the cluster should be able to provide a similar interface. The problem is other than the PD, other nodes have no similar method to get the information of all nodes locally. So it seems reasonable to provide a list of all nodes by the client.